spikingjelly/activation_based/triton_kernel/flex_sn_inductor/hop.py

```
@@ -143,3 +668,151 @@ def eager_scan(
+                    "flex_sn_scan only supports tensor inputs and states"
+                )
+
+            from torch._dynamo.variables.constant import ConstantVariable as _ConstantVariable
+
+            step_inputs = [
+                arg.call_method(tx, "__getitem__", [_ConstantVariable(0)], {})
+                for arg in flat_args[:num_inputs]
+            ]
+            body_args = [*step_inputs, *flat_args[num_inputs:]]
+
+            if "source_target" in getattr(speculate_subgraph, "__code__").co_varnames:
```

Ruff (0.15.10)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)